parachain/src/xcm.rs

```diff
@@ -0,0 +1,238 @@
+		X4(Junction, Junction, Junction, Junction),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum Junction {
+		Parent,
+		Parachain { #[codec(compact)] id: u32 },
+		OpaqueRemark(Vec<u8>),
+		AccountId32 { network: MultiNetwork, id: [u8; 32] },
+		AccountIndex64 { network: MultiNetwork, #[codec(compact)] index: u64 },
+		AccountKey20 { network: MultiNetwork, key: [u8; 20] },
+		/// An instanced Pallet on a Frame-based chain.
+		PalletInstance { id: u8 },
+		/// A nondescript index within the context location.
+		GeneralIndex { #[codec(compact)] id: u128 },
+		/// A nondescript datum acting as a key within the context location.
+		GeneralKey(Vec<u8>),
+	}
+
+	impl From<Junction> for MultiLocation {
+		fn from(x: Junction) -> Self {
+			MultiLocation::X1(x)
+		}
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum AssetInstance {
+		Undefined,
+		Index8(u8),
+		Index16 { #[codec(compact)] id: u16 },
+		Index32 { #[codec(compact)] id: u32 },
+		Index64 { #[codec(compact)] id: u64 },
+		Index128 { #[codec(compact)] id: u128 },
+		Array4([u8; 4]),
+		Array8([u8; 8]),
+		Array16([u8; 16]),
+		Array32([u8; 32]),
+		Blob(Vec<u8>),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum MultiAsset {
+		Wild,
+		WildFungible,
+		WildNonFungible,
+		WildAbstractFungible { id: Vec<u8> },
+		WildAbstractNonFungible { class: Vec<u8> },
+		WildConcreteFungible { id: MultiLocation },
+		WildConcreteNonFungible { class: MultiLocation },
+		AbstractFungible { id: Vec<u8>, #[codec(compact)] amount: u128 },
+		AbstractNonFungible { class: Vec<u8>, instance: AssetInstance },
+		ConcreteFungible { id: MultiLocation, #[codec(compact)] amount: u128 },
+		ConcreteNonFungible { class: MultiLocation, instance: AssetInstance },
+		Each(Vec<MultiAsset>),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+	pub enum Ai {
+		Each(Vec<Ai>),
+		DepositAsset { asset: MultiAsset, dest: MultiLocation },
+		ExchangeAsset { give: MultiAsset, receive: MultiAsset },
+		InitiateReserveTransfer { asset: MultiAsset, dest: MultiLocation, effect: Box<Ai> },
+		InitiateTeleport { asset: MultiAsset, dest: MultiLocation, effect: Box<Ai> },
+		QueryHolding { #[codec(compact)] query_id: u64, dest: MultiLocation, assets: Vec<MultiAsset> },
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+	pub enum Xcm {
+		WithdrawAsset { asset: MultiAsset, effect: Ai },
+		// Equivalent to WithdrawAsset{asset, Ai::InitiateReserveTransfer{asset, dest, effect}
+		ReserveAssetTransfer { asset: MultiAsset, dest: MultiLocation, effect: Ai },
+		ReserveAssetCredit { asset: MultiAsset, effect: Ai },
+		TeleportAsset { asset: MultiAsset, effect: Ai },
+		Balances { query_id: Vec<u8>, assets: Vec<MultiAsset> },
+		Transact { origin_type: MultiOrigin, call: Vec<u8> },
+		// these won't be staying here for long. only v0 parachains with HRMP.
+		ForwardToParachain { id: u32, inner: Box<VersionedXcm> },
```

Comment:
Not sure if it is possible to set explicit index here to 250 so remove those in later version won't leave a hole